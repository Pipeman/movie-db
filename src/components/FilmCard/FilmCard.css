.movie-card-container {
  text-align: center;
}

.movie-card-container h1 {
  font-size: 24px;
}

.movie-card-container header {
  text-align: left;
}

.poster {
  position: relative;
}

.poster-not-available {
  width: 185px;
  height: 278px;
  line-height: 278px;
  background-color: #dbdbdb;
}