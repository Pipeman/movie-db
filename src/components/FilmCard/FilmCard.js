import React from "react";
import { IMAGE_BASE_URL } from "../../config/config";
import styles from "./FilmCard.css"

const FilmCard = ({ info }) => (
  <div className={styles["movie-card-container"]}>
    {info.poster_path
      ? <img className={styles["poster"]} src={IMAGE_BASE_URL + info.poster_path}/>
      : <div className={styles["poster-not-available"]}></div>}
    <header>
      <h1>{info.title}</h1>
      <span>{info.vote_average} over {info.vote_count} votes</span>
    </header>
  </div>
);

export default FilmCard;