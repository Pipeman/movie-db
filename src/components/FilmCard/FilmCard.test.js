import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import FilmCard from "./FilmCard";

let wrapper;
const info = {
    title: "Fight Club",
    vote_average: 10.0,
    vote_count: 1000,
    poster_path: "fight-club-poster.jpg",
};

beforeEach(() => {
  wrapper = mount(
    <FilmCard
        info={info}
    />
  );
});

it("should render the component correctly", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});
