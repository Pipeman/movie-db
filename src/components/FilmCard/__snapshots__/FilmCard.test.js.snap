// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly 1`] = `
<FilmCard
  info={
    Object {
      "poster_path": "fight-club-poster.jpg",
      "title": "Fight Club",
      "vote_average": 10,
      "vote_count": 1000,
    }
  }
>
  <div>
    <img
      src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/fight-club-poster.jpg"
    />
    <header>
      <h1>
        Fight Club
      </h1>
      <span>
        10
         over 
        1000
         votes
      </span>
    </header>
  </div>
</FilmCard>
`;
